geopandas/io/tests/test_io.py

```diff
@@ -32,19 +32,34 @@ def test_read_postgis_default(self):
 
     def test_read_postgis_custom_geom_col(self):
         con = connect('test_geopandas')
-        if con is None or not create_db(self.df):
+        geom_col = "the_geom"
+        if con is None or not create_db(self.df, geom_col=geom_col):
             raise pytest.skip()
 
         try:
-            sql = """SELECT
-                     borocode, boroname, shape_leng, shape_area,
-                     geom AS __geometry__
-                     FROM nybb;"""
-            df = read_postgis(sql, con, geom_col='__geometry__')
```

Comment:
This kind of query no longer would work, because of looking for an `srid` associated with `"__geometry__"`. This test seemed mostly like a convoluted way to verify that alternative column names worked, rather than an intended feature, but making `AS` not work on the geometry column might be more undesirable than I'm appreciating.

Reply:
Yes, I think renaming a column with 'AS' is very common and would be unfortunate if that failed

Reply:
Fair enough. Given my understanding of what's out there, even incorporating a dependency on `geoalchemy2` it seems like we'd probably need either significant expansion of how much sql handling we're doing in geopandas, or we'll have to make some small tradeoffs in terms of limiting flexibility to offer up an improved default experience. This exercise started as an attempt just to see what was possible and where problems came up, and maybe help move the conversation in #595 forward in one direction or another.